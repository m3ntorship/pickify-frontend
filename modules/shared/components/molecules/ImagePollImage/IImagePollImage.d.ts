import type { IVoteIcon } from '../../atoms/VoteIcon/IVoteIcon';

declare namespace IImagePollImage {
  export interface IProps extends IVoteIcon.IProps {
    imageUrl: string;
  }
}
export { IImagePollImage };
