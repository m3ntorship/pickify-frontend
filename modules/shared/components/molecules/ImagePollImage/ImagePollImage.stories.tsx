import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import ImagePollImage from './ImagePollImage';
import type { IImagePollImage } from './IImagePollImage';

export default {
  component: ImagePollImage,
  title: 'Molecules/ImagePollImage',
};
const Template: Story<IImagePollImage.IProps> = (args): ReactElement => (
  <ImagePollImage {...args} />
);
export const Default = Template.bind({});
Default.args = {};
