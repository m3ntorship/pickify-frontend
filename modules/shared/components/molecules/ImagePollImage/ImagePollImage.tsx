import React from 'react';
import type { FC, ReactElement } from 'react';
import Image from 'next/image';
import type { IImagePollImage } from './IImagePollImage';
import VoteIcon from '../../atoms/VoteIcon/VoteIcon';

const ImagePollImage: FC<IImagePollImage.IProps> = ({
  isOneImageVote,
  imageUrl,
}): ReactElement => {
  return (
    <div className="relative w-auto">
      <Image
        src={imageUrl}
        layout="fill"
        objectFit="contain"
        className="w-full h-full"
      />
      <div className="absolute bottom-4 right-4">
        <VoteIcon isOneImageVote={isOneImageVote} />
      </div>
      <div className="absolute bottom-4 left-4">
        <p>image caption</p>
      </div>
    </div>
  );
};

export default ImagePollImage;
