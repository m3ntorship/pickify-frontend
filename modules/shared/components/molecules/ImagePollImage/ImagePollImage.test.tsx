import * as React from 'react';
import * as renderer from 'react-test-renderer';
import ImagePollImage from './ImagePollImage';

describe('Privacy Component with snapshot', () => {
  it('should render the like and dislike voteIcon  ', () => {
    const tree = renderer
      .create(
        <ImagePollImage
          isOneImageVote
          imageUrl="https://source.unsplash.com/random"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render the heart voteIcon  ', () => {
    const tree = renderer
      .create(
        <ImagePollImage
          isOneImageVote={false}
          imageUrl="https://source.unsplash.com/random"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
