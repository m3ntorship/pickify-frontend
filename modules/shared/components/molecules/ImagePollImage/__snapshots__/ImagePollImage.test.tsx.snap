// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Privacy Component with snapshot should render the heart voteIcon   1`] = `
<div
  className="relative w-auto"
>
  <div
    style={
      Object {
        "bottom": 0,
        "boxSizing": "border-box",
        "display": "block",
        "left": 0,
        "margin": 0,
        "overflow": "hidden",
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <img
      className="w-full h-full"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": "contain",
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "visibility": "hidden",
          "width": 0,
        }
      }
    />
  </div>
  <div
    className="absolute bottom-4 right-4"
  >
    <div
      className="flex justify-center"
    >
      <button
        className="love group"
        type="button"
      >
        <svg
          className="fill-primary stroke-current stroke-2 text-white group-hover:fill-white"
          data-file-name="SvgFilledHeart"
        />
      </button>
    </div>
  </div>
  <div
    className="absolute bottom-4 left-4"
  >
    <p>
      image caption
    </p>
  </div>
</div>
`;

exports[`Privacy Component with snapshot should render the like and dislike voteIcon   1`] = `
<div
  className="relative w-auto"
>
  <div
    style={
      Object {
        "bottom": 0,
        "boxSizing": "border-box",
        "display": "block",
        "left": 0,
        "margin": 0,
        "overflow": "hidden",
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <img
      className="w-full h-full"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": "contain",
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "visibility": "hidden",
          "width": 0,
        }
      }
    />
  </div>
  <div
    className="absolute bottom-4 right-4"
  >
    <div
      className="flex justify-center"
    >
      <div
        className="flex"
      >
        <button
          className="dislike group"
          type="button"
        >
          <svg
            className="fill-error stroke-current stroke-1 text-white group-hover:fill-white"
            data-file-name="SvgFilledDislike"
          />
        </button>
        <button
          className="like group"
          type="button"
        >
          <svg
            className="fill-primary stroke-current stroke-1 text-white group-hover:fill-white"
            data-file-name="SvgFilledLike"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="absolute bottom-4 left-4"
  >
    <p>
      image caption
    </p>
  </div>
</div>
`;
